### DISCUSSION TITLE
TLS 1.0, on an old program that needs to download a file

### DISCUSSION THREAD

**Discussion by shadvalentine** [community/community#60657] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Feature Area

Issues

### Body

I have a plugin (GoArrow) for an old game (Asheron's Call) that is no longer being updated, and I do not have access to source code.  It uses TLS 1.0 to download updates, with a configurable URL.  I would like to use GitHub for this file, as it is in xml format, and very easy to edit on GitHub, however the plugin will not download the file from GitHub.  I am assuming that it is due to it being old, and another application was able to allow it to download as soon as I enabled TLS 1.0 on it.  Is there any way to get around the TLS issue, like turning off SSL for the file in GitHub?

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by jussihi** [comment]
Be sure to use the link to the raw file on github.

For example in my project, raw link for the file https://github.com/jussihi/SMM-Rootkit/blob/master/README.md would be https://raw.githubusercontent.com/jussihi/SMM-Rootkit/master/README.md

**Reply by shadvalentine** [reply]
Thank you for your response.  I am using the raw file link for it, unfortunately it does not allow the download, being the plugin is so old and uses TLS 1.0.

**Reply by jussihi** [reply]
Does the plugin give any error when trying to download the file?

**Reply by shadvalentine** [reply]
Just that the connection was closed, or refused, cannot remember exactly.  I did search the error when I tested it before and it talked about the TLS error.  I was able to confirm the plugin is using TLS 1.0 with the other program, as it showed the refused connection due to TLS 1.0.  GetHub currently does not allow TLS 1.0 connection by default, the other program has the ability to change settings to allow it, I cannot find an option like that in GutHub.

**Reply by jussihi** [reply]
Ok. In that case I am afraid that there is no way to make it work. Github indeed enforces TLS 1.2 already. And there is no way to remove the encryption (choosing HTTP instead of HTTPS) on GitHub. The only way I can imagine is to use a 3rd-party webhost with TLS 1.3 capability and use it as a "proxy" to the Github page. I.e., whenever a user requests the file on your 3rd-party webhost (with old TLS or no encryption at all), it requests the page from github with TLS 1.3 and forwards the contents to your game client.

I know this is not a solution with using only GitHub, but I think this would be the easiest way if you want to keep the version control and do updates on github.

**Reply by shadvalentine** [reply]
That is a method I looked at briefly, do you know how this could be done?  My first attempt was to use an php include, with no luck.  That was on an old site, and may still be using TLS 1.0, so that could be a reason it did not work.

**Reply by jussihi** [reply]
Now thinking of it, another way would be to create a python script that you run locally and that does the "proxying". Then you would not need a 3rd party webhost running 😃

**Reply by shadvalentine** [reply]
Only issue with that method is that I need this to be accessible by many other people that play on the server.  That would require each one of them to run the script on their machine.

**Reply by jussihi** [reply]
In that case I would use a ready-made codebase like https://github.com/jenssegers/php-proxy . The example seems to suit your use case fairly well, just change the `example.com` to your xml file location.

**Reply by shadvalentine** [reply]
Thank you for that, I will check it out tomorrow and get back to you.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @shadvalentine ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
